cli/commands/create_benchmark_pipeline_command.py

```diff
@@ -0,0 +1,164 @@
+import humps
+from configargparse import ArgParser
+from pathlib import Path
+from string import Template
+
+from mcs_benchmark_data.path import DATA_DIR_PATH, TEST_DATA_DIR_PATH
+from mcs_benchmark_data.dataset_type import DatasetType
+from mcs_benchmark_data.template_type import TemplateType
+
+from cli.commands._command import _Command
+
+
+class CreateBenchmarkPipelineCommand(_Command):
+    def add_arguments(self, arg_parser: ArgParser, add_parent_args):
+        arg_parser.add_argument(
+            "--benchmark_name", help="name of the new benchmark (in snake_case)"
+        )
+        arg_parser.add_argument(
+            "--using-test-data",
+            help="true if using truncated data for testing (in the test_data directory)\nalters the test file input path",
+        )
+
+    def __call__(self, args):
+
+        root_path = Path(__file__).parent.parent.parent
+
+        benchmark_name = args.benchmark_name
+
+        using_test_data = args.using_test_data
+
+        if not humps.is_snakecase(benchmark_name):
+            raise ValueError(
+                "This benchmark name is not in snake_case. See https://medium.com/better-programming/string-case-styles-camel-pascal-snake-and-kebab-case-981407998841 for more information."
+            )
+
+        if using_test_data:
+            data_dir = "TEST_DATA_DIR_PATH"
+        else:
+            data_dir = "DATA_DIR_PATH"
+
+        self.__make_benchmark_directories(
+            root_path=root_path, benchmark_name=benchmark_name
+        )
+
+        self.__create_files_from_template(
+            root_path=root_path, benchmark_name=benchmark_name, data_dir=data_dir
+        )
+
+    def __create_files_from_template(
+        self, root_path: Path, benchmark_name: str, data_dir: str
+    ):
+
+        path_to_templates = root_path / "templates"
+
+        format_args = {
+            "benchmark_name": benchmark_name,
+            "BenchmarkName": humps.pascalize(benchmark_name),
+            "data_dir": data_dir,
+        }
+
+        for template_type in TemplateType:
+
+            template_module = importlib.import_module(
+                f"mcs_benchmark_data.{template_type.value}_template_metadata"
+            )
+            TemplateDataclass = getattr(
+                template_module, f"{template_type.value.capitalize()}TemplateMetadata"
+            )
+
+            template_metadata = TemplateDataclass(benchmark_name=benchmark_name)
+
+            with open(
+                path_to_templates / template_metadata.template_name
+            ) as template_file:
+                template_str = template_file.read()
+
+            if template_type == TemplateType.METADATA:
+                temp = Template(template_str)
+                formatted_str = temp.substitute(**format_args)
+
+            else:
+                formatted_str = template_str.format(**format_args)
+
+            with open(
+                root_path / template_metadata.dest_file_path_from_root, "w"

```

Comment:
"w" won't overwrite, but you probably want to check that the file doesn't exist and raise an exception yourself if it does.